File: prog-front/src/lib/types/room.ts
import { Building } from "./building";

// Tipos de aulas/salas
export interface Room {
  id: number;
  building_id: number;
  building?: Building;
  name: string;
  capacity: number;
  resources?: string | object; // JSONB
  description?: string;
  created_at: string;
  updated_at: string;
}

File: prog-back/internal/models/room.go
package models

import "time"

type Room struct {
	ID          uint      `gorm:"primaryKey" json:"id"`
	BuildingID  uint      `gorm:"index;not null" json:"building_id"`
	Building    *Building `gorm:"foreignKey:BuildingID" json:"building,omitempty"`
	Name        string    `gorm:"not null" json:"name"`
	Capacity    int       `gorm:"not null" json:"capacity"`
	Resources   string    `json:"resources"` // JSON string or comma-separated
	Description string    `json:"description"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}
